File: LeetCode/Easy/0283-move-zeroes/0283-move-zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        var insertIdx = 0;

        for (var i = 0; i < nums.Length; i++)
        {
            if (nums[i] != 0)
            {
                if (insertIdx != i)
                {
                    nums[insertIdx] = nums[i];
                }
                insertIdx++;
            }
        }
        
        for (var i = insertIdx; i < nums.Length; i++)
        {
            nums[i] = 0;
        }
    }
}

File: LeetCode/Easy/0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        write_idx = 0
        for n in nums:
            if n != 0:
                nums[write_idx] = n
                write_idx += 1

        for i in range(write_idx, len(nums)):
            nums[i] = 0